cdap-app-fabric/src/main/java/io/cdap/cdap/security/impersonation/DefaultUGIProvider.java

```diff
@@ -172,4 +202,59 @@ private File localizeKeytab(Location keytabLocation) throws IOException {
 
     return localKeytabFile.toFile();
   }
+
+  /**
+   * Returns the runtime user supplied arguments to a program if entity is of type ProgramRun
+   *
+   * @param programId the entity id
+   * @return properties map
+   */
+  private Map<String, String> getRuntimeProperties(NamespacedEntityId programId) {
+    if (programId instanceof ProgramRunId) {
+      ProgramRunId runId = ((ProgramRunId) programId);
+      RunRecordMeta runRecord = null;
+
+      long sleepDelayMs = TimeUnit.MILLISECONDS.toMillis(100);
+      long timeoutMs = TimeUnit.SECONDS.toMillis(10);
+
+      try {
+        runRecord = Retries.callWithRetries(() -> {
+          try {
+            RunRecordMeta rec = store.getRun(runId);
+            if (rec != null) {
+              return rec;
+            }
+          } catch (Exception e) {
+            LOG.debug("Exception while trying to fecth RunRecord %s, retrying again: ", e.getMessage());
```

Comment:
Better not to catch and log here. Just let the exception propagate. The `Retries` class already has a sampling logger to log this.

Reply:
Fixed.